VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/FlagsEnumValuesAreNotPowersOfTwo/FlagsEnumValuesAreNotPowersOfTwoAnalyzer.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace VSDiagnostics.Diagnostics.General.FlagsEnumValuesAreNotPowersOfTwo
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class FlagsEnumValuesAreNotPowersOfTwoAnalyzer : DiagnosticAnalyzer
+    {
+        private const string DiagnosticId = nameof(FlagsEnumValuesAreNotPowersOfTwoAnalyzer);
+        private const DiagnosticSeverity Severity = DiagnosticSeverity.Error;
+
+        private static readonly string Category = VSDiagnosticsResources.GeneralCategory;
+        private static readonly string Message = VSDiagnosticsResources.FlagsEnumValuesAreNotPowersOfTwoAnalyzerMessage;
+        private static readonly string Title = VSDiagnosticsResources.FlagsEnumValuesAreNotPowersOfTwoAnalyzerTitle;
+
+        internal static DiagnosticDescriptor Rule => new DiagnosticDescriptor(DiagnosticId, Title, Message, Category, Severity, true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzeSymbol, SyntaxKind.EnumDeclaration);
+        }
+
+        private void AnalyzeSymbol(SyntaxNodeAnalysisContext context)
+        {
+            var declarationExpression = context.Node as EnumDeclarationSyntax;
+            if (declarationExpression == null)
+            {
+                return;
+            }
+
+            if (!declarationExpression.AttributeLists.Any(
+                    a => a.Attributes.Any(
+                        t => context.SemanticModel.GetSymbolInfo(t).Symbol.ContainingType.MetadataName == typeof(FlagsAttribute).Name)))
```

Comment:
I would expect that this might cause a problem when manually testing: first define the enum and then add the `[Flags]` attribute. If you type slowly enough, it will try to get the symbol of the unfinished attribute (say.. `[Fla`), trigger a symbol lookup, return null because its unfinished code and then cause a NRE. Can you test this out to be sure?